### PR TITLE
coro_file add unit test

### PR DESCRIPTION
**Pull request by zhengjian526**
Add unit tests for reading and writing files, including large files, small files and empty files.

### CODE REVIEW

File: src/coro_file/tests/test_corofile.cpp [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+
+  if (!file) {
+    std::cout << "create file failed\n";
+    return std::vector<char>{};
+  }
+  size_t fill_with_size = fill_with.size();
+  if (file_size == 0 || fill_with_size == 0) {
+    return std::vector<char>{};
+  }
+  std::vector<char> ret{create_filled_vec(fill_with, file_size)};
```
Comment (qicosmos) [inline src/coro_file/tests/test_corofile.cpp:47]: 这里可以通过循环方式去写文件，没必要返回vector，否则如果你要测试一个10G的文件内存可能都溢出了。

比如你循环写
```c++
std::vector<char> v(1024, 'a');
for(...)
  file.write(v.data(), v.size());
```

这样写非常大的文件都没有内存问题。
